### PR TITLE
Switch to Azure for GPU CI

### PR DESCRIPTION
**Pull request by drasmuss**
Makes the GPU builds more reliable and less prone to conflicts.

### CODE REVIEW

File: .templates/gpu.sh.template [hunk 1]
```diff
@@ -14,14 +29,9 @@
         (
             flock -x -w 540 200 || exit 1
             export CUDA_VISIBLE_DEVICES="$GPU_NUM"
-            export TF_FORCE_GPU_ALLOW_GROWTH=true
             pytest $TEST_ARGS nengo_dl/tests/test_benchmarks.py::test_performance --performance -v --durations 20 --color=yes || exit 1
+            export TF_FORCE_GPU_ALLOW_GROWTH=true
```
Comment (hunse) [inline .templates/gpu.sh.template]: Do we need this at all? What's the problem with taking over the whole GPU?

File: nengo_dl/tests/test_benchmarks.py [hunk 2]
```diff
@@ -196,10 +196,10 @@ def test_lmu(Simulator, native_nengo, pytestconfig):
 @pytest.mark.parametrize(
     "net, train, minibatch_size, min, max",
     [
-        (benchmarks.cconv(128, 64, nengo.RectifiedLinear()), False, 64, 0.6, 0.75),
-        (benchmarks.cconv(128, 64, nengo.LIF()), False, 64, 1.4, 1.6),
-        (benchmarks.integrator(128, 32, nengo.RectifiedLinear()), True, 64, 0.55, 0.9),
-        (benchmarks.integrator(128, 32, nengo.LIF()), True, 64, 0.95, 1.25),
+        (benchmarks.cconv(128, 64, nengo.RectifiedLinear()), False, 64, 0.95, 1.1),
```
Comment (hunse) [inline nengo_dl/tests/test_benchmarks.py]: Would it be possible to keep the benchmarks from `abrgpu` around somewhere? It might be nice to have a command-line pytest option that specifies where we're running, and chooses the appropriate numbers. When we do a release, we might want to run the benchmarks on both `abrgpu` and Azure just to make sure they haven't regressed in either location.

File: .templates/gpu.sh.template (hunk 1 above)
Comment (drasmuss) [reply .templates/gpu.sh.template]: It's because we're running the tests in two processes to speed things up (so those processes need to be able to share the GPU).

File: nengo_dl/tests/test_benchmarks.py (hunk 2 above)
Comment (drasmuss) [reply nengo_dl/tests/test_benchmarks.py]: To be honest, I suspect we wouldn't get a lot out of occasionally running things on `abrgpu`. I'm just picturing if we did detect a regression on `abrgpu` that hadn't shown up on Azure, I think 99% of the time we would just chalk it up to environmental differences and either ignore it or spend a bunch of time fixing the `abrgpu` environment (rather than touching any code in Nengo DL). We'd also need to maintain both sets of benchmarks as we go, which is a pain (whenever we need to adjust these benchmarks it sucks up a bunch of time rerunning the builds over and over and tweaking the numbers).

So, long story short, it'd be great to have a more robust regression testing framework that was regularly testing against a bunch of different environments. But I wouldn't dedicate the time to doing a partial version of that now given our resources.

Comment (hunse) [reply nengo_dl/tests/test_benchmarks.py]: That makes sense. My main concern is that the GPUs on Azure are pretty crappy, and not that representative of what we'll typically be running on. But you're right that the regression tests aren't great themselves, and perhaps also not fully representative of a larger end-to-end training, which is where we more care about performance.
